File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/profile/address/ui/AddressPresenterModule.java
package com.fintechplatform.ui.profile.address.ui;

import com.fintechplatform.ui.models.DataAccount;
import com.fintechplatform.api.profile.api.ProfileAPI;
import com.fintechplatform.ui.profile.db.user.UsersPersistanceDB;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AddressPresenterModule {

    private AddressContract.View view;
    private DataAccount dataAccount;

    public AddressPresenterModule(AddressContract.View view, DataAccount dataAccount) {
        this.view = view;
        this.dataAccount = dataAccount;
    }

    @Provides
    @Singleton
    AddressContract.Presenter providesAddressPresenter(ProfileAPI api, UsersPersistanceDB userPersistanceDB) {
        return new AddressPresenter(view, api, dataAccount, userPersistanceDB);
    }
}

File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/profile/address/di/AddressPresenterModule.kt
package com.fintechplatform.ui.profile.address.di

import com.fintechplatform.api.profile.api.ProfileAPI
import com.fintechplatform.ui.models.DataAccount
import com.fintechplatform.ui.profile.address.AddressContract
import com.fintechplatform.ui.profile.address.AddressPresenter
import com.fintechplatform.ui.profile.db.user.UsersPersistanceDB
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AddressPresenterModule(private val view: AddressContract.View, private val dataAccount: DataAccount) {

    @Provides
    @Singleton
    internal fun providesAddressPresenter(api: ProfileAPI, userPersistanceDB: UsersPersistanceDB): AddressContract.Presenter {
        return AddressPresenter(view, api, dataAccount, userPersistanceDB)
    }
}